### PR TITLE
MLPAB-66 - Create a load balancer and DNS for app

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Send requests from users to ECS

Fixes MLPAB-66 MLPAB-16

## Approach

- Create an application load balancer
- Create ingress security groups
- Define an allow list of ip ranges for non production environments (though production will be behind this list also)
- Create DNS record for app

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* ~I have added tests to prove my work~
* ~I have added welsh translation tags and updated translation files~
* ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
* ~The product team have tested these changes~

### CODE REVIEW

File: terraform/environment/region/modules/app/alb.tf [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#   from_port         = 443
+#   to_port           = 443
+#   protocol          = "tcp"
+#   cidr_blocks       = ["0.0.0.0/0"] #tfsec:ignore:aws-vpc-no-public-ingress-sgr - open ingress for production
+#   security_group_id = aws_security_group.app_loadbalancer.id
+#   provider          = aws.region
+# }
+
+resource "aws_security_group_rule" "app_loadbalancer_egress" {
+  description       = "Allow any egress from Use service load balancer"
```
Comment (acsauk) [inline terraform/environment/region/modules/app/alb.tf:126]: Is this `Use` as in Use an LPA or some other `Use`?

**Review by acsauk — APPROVED** [review]
All looks sensible to me - one Q on the description of a resource 👍

File: terraform/environment/region/modules/app/alb.tf (hunk 1 above)
Comment (andrewpearce-digital) [reply terraform/environment/region/modules/app/alb.tf]: well spotted, removed that

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**